### PR TITLE
Add admin dashboard - manage mentor applications scene

### LINKED ISSUE #26: Implement the admin dashboard - manage mentor applications view
### Description
As part of this issue, we need to create the admin dashboard - manage mentor applications view, which should closely match the provided mockup. This will provide the admins with a way to manage all the mentor applications with the functionalities for searching, filtering or changing status.

### Tasks
1. Set up the basic React components as required.
1. Implement the layout and design of the admin dashboard - manage mentor applications view, following the provided mockup.
1. Integrate the necessary components and assets to achieve the desired functionality and visual appearance.
1. Ensure responsiveness and compatibility across different devices and screen sizes.

Mockup: https://www.figma.com/file/nehm2ff8aR7NkPerg39fK9/ScholarX-V2-%7C-Wireframes?type=design&node-id=1974-2421&mode=design&t=zBGgiYyW1foSirHq-4

image

Note: You may want to mock the API refer to the [API doc](https://documenter.getpostman.com/view/27421496/2s93m1a4ac#d708a4b2-3d26-4089-9a27-d9d71d449f62) to see the relevant details.

### Acceptance Criteria
- The user interface closely resembles the provided mockup.
- All required sections, elements, and content are present and displayed correctly.
- The interface is responsive and works well on various devices and screen sizes.
- The implementation follows best practices and coding standards.
- Unit tests are written.

### PR DESCRIPTION
**Pull request by Madhawa97**
## Purpose

The purpose of this PR is to fix #26

## Goals

This will provide the admins with a way to manage all the mentor applications with the functionalities for searching, filtering or changing status.

## Approach

Added changes according to the UI mockups and wireframes.

## Screenshots
image

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/components/Dashboard/scenes/ManageMentorApplications/ManageMentorApplications.tsx [hunk 1]
```diff
@@ -1,11 +1,223 @@
...
+        { withCredentials: true }
+      );
+      if (response.status === 200) {
+        void fetchMentors();
+      } else {
+        console.error('Failed to update mentor status');
+      }
+    } catch (error) {
+      console.error('Error updating mentor status:', error);
+    }
```
Comment (anjula-sack) [inline src/components/Dashboard/scenes/ManageMentorApplications/ManageMentorApplications.tsx]: use react query here. https://tanstack.com/query/latest/docs/framework/react/guides/queries

File: src/components/Dashboard/scenes/ManageMentorApplications/ManageMentorApplications.tsx [hunk 2]
```diff
@@ -1,11 +1,223 @@
...
+    const url = `${API_URL}/admin/mentors`;
+    try {
+      const response = await axios.get(url, { withCredentials: true });
+      setMentors(response.data.mentors);
+    } catch (error) {
+      console.error('Error fetching mentors:', error);
+    } finally {
+      setLoading(false);
+    }
+  };
```
Comment (anjula-sack) [inline src/components/Dashboard/scenes/ManageMentorApplications/ManageMentorApplications.tsx]: create a custom hook for this useMentors and use useQuery to fetch them with axios. take a look at the mentor application.

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/components/Dashboard/scenes/ManageMentorApplications/ManageMentorApplications.tsx (hunk 1 above)
Comment (Madhawa97) [reply src/components/Dashboard/scenes/ManageMentorApplications/ManageMentorApplications.tsx]: Thanks! First-time case across this. seems super awesome.

File: src/components/Dashboard/scenes/ManageMentorApplications/ManageMentorApplications.tsx (hunk 2 above)
Comment (Madhawa97) [reply src/components/Dashboard/scenes/ManageMentorApplications/ManageMentorApplications.tsx]: Implemented a custom hook `useMentors` as requested. please check.

File: src/components/Dashboard/scenes/ManageMentorApplications/ManageMentorApplications.tsx [hunk 3]
```diff
@@ -1,11 +1,202 @@
...
+    );
+  };
+
+  const renderMentorTable = () => {
+    const filteredMentors =
+      mentors != null
+        ? filter !== ''
+          ? mentors.filter((mentor) => mentor.state === filter)
+          : mentors
+        : [];
```
Comment (anjula-sack) [inline src/components/Dashboard/scenes/ManageMentorApplications/ManageMentorApplications.tsx:83]: why can't we use this like this
`const filteredMentors = mentors?.filter(mentor => mentor.state === filter) || [];`

**Review by anjula-sack — APPROVED** [review]
Thank you! @Madhawa97

**Comment by Madhawa97** [comment]
Its a pleasure @anjula-sack

**Pull request merged**